### PR TITLE
[QSP-1]  fix: add update staking parameter transition

### LINKED ISSUE #24: QSP-1 Staking Parameters May Lead to an Incorrect State
Description: There is no sanity check to ensure staking-related parameters are set correctly, i.e., that minstake <= maxstake and maxstake <=
contractmaxstake always hold.

Recommendation: Remove all transitions setting each staking parameter individually. Then, have a transition to set all three staking variables at once, checking that the invariant minstake <= maxstake and maxstake <= contractmaxstake holds.

### PR DESCRIPTION
**Pull request by renlulu**
This pr is to fix #24.

### CODE REVIEW

File: contracts/multisig_wallet.scilla [hunk 1]
```diff
@@ -77,19 +77,16 @@ type ProxyTransaction =
...
 | UndateVerifier of ByStr20
 (* drain_contract_balance ()  *)
 | DrainContractBalance
-(* update_minstake (min_stake : Uint128) *)
-| UpdateMinStake of Uint128
-(* update_maxstake (max_stake : Uint128) *)
-| UpdateMaxStake of Uint128
-(* update_contractmaxstake (max_stake : Uint128) *)
-| UpdateContractMaxStake of Uint128
+(* update_staking_parameter (minx_stake : Uint128, max_stake : Uint128, contract_max_stake : Uint128) *)
```
Comment (bb111189) [inline contracts/multisig_wallet.scilla:88]: I think there is a typo . `minx_stake` should be `min_stake`

File: contracts/ssnlist.scilla [hunk 2]
```diff
@@ -268,34 +279,35 @@ transition drain_contract_balance (initiator : ByStr20)
...
-transition update_minstake (min_stake : Uint128, initiator : ByStr20)
-  is_paused;
-  validate_proxy;
-  validate_admin initiator;
-  minstake := min_stake
-end
-
-(* @dev: Set the maxstake of contract. Used by admin only. *)
 (* @param max_stake: New maxstake value *)
+(* @param max_stake: New contract maxstake value *)
```
Comment (bb111189) [inline contracts/ssnlist.scilla:285]: should be `contract_max_stake`

File: contracts/ssnlist.scilla [hunk 3]
```diff
@@ -268,34 +279,35 @@ transition drain_contract_balance (initiator : ByStr20)
...
+    match less_than with
+    | True =>
+      minstake := min_stake;
+      maxstake := max_stake;
+      contractmaxstake := contract_max_stake;
+      e = mk_update_staking_parameter_event initiator;
+      event e
+    | False => 
+      e = mk_update_staking_parameter_error initiator min_stake max_stake contract_max_stake;
+      event e
```
Comment (bb111189) [inline contracts/ssnlist.scilla:304]: should we throw exception here?

File: contracts/ssnlist.scilla [hunk 4]
```diff
@@ -268,34 +279,35 @@ transition drain_contract_balance (initiator : ByStr20)
...
+    end
+  | False =>
+     e = mk_update_staking_parameter_error initiator min_stake max_stake contract_max_stake;
+     event e
```
Comment (bb111189) [inline contracts/ssnlist.scilla:308]: should we throw an exception here?

File: contracts/multisig_wallet.scilla (hunk 1 above)
Comment (renlulu) [reply contracts/multisig_wallet.scilla:88]: Ok, let me fix it.

**Pull request merged**